app/domain/authentication/authn_jwt/vendor_configurations/configuration_jwt_generic_vendor.rb

```diff
@@ -5,29 +5,27 @@ module VendorConfigurations
       class ConfigurationJWTGenericVendor < ConfigurationInterface
 
         def initialize(authenticator_input)
-          @authentication_parameters_class = Authentication::AuthnJwt::AuthenticationParameters
-          @extract_token_from_credentials = Authentication::AuthnJwt::InputValidation::ExtractTokenFromCredentials.new
-          @validate_and_decode_token_class = Authentication::AuthnJwt::ValidateAndDecode::ValidateAndDecodeToken
-          @identity_provider_factory = Authentication::AuthnJwt::IdentityProviders::CreateIdentityProvider
-          @validate_resource_restrictions_class = Authentication::ResourceRestrictions::ValidateResourceRestrictions
-          @create_identity_provider_class = Authentication::AuthnJwt::IdentityProviders::CreateIdentityProvider
           @logger = Rails.logger
-          init_authentication_parameters(authenticator_input)
+
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CREATING_AUTHENTICATION_PARAMETERS_OBJECT.new)
+          @authentication_parameters = authentication_parameters_class.new(
+            authentication_input: authenticator_input,
+            jwt_token: jwt_token(authenticator_input)
+          )
         end
 
         def jwt_identity
           @jwt_identity ||= jwt_identity_from_request
         end
 
         def validate_restrictions
-          initialize_validate_restrictions
-          @validate_resource_restrictions.call(
+          validate_resource_restrictions.call(
             authenticator_name: @authentication_parameters.authenticator_name,
             service_id: @authentication_parameters.service_id,
             account: @authentication_parameters.account,
             role_name: jwt_identity,
-            constraints: @constraints,
-            authentication_request: @restriction_validator.new(
+            constraints: constraints,
+            authentication_request: restriction_validator_class.new(
               decoded_token: @authentication_parameters.decoded_token
             )
```

Comment:
is this correct? why does the `restriction_validator_class` return an `authentication_request` object?

Reply:
ValidateResourceRestrictions recivies the validator as authentication_request
https://github.com/cyberark/conjur/blob/ba3a03e841eebe427b340bda5cf5983be1029f6e/app/domain/authentication/resource_restrictions/validate_resource_restrictions.rb#L28

I also don't agree with the naming but don't think its in scope for the jwt authenticator to fix this for all authenticators.